meson.build

```diff
@@ -537,6 +537,12 @@ endforeach
 ############################################################
 
 conf.set_quoted('FALLBACK_HOSTNAME', get_option('fallback-hostname'))
+gateway_hostnames = get_option('gateway-hostnames').split(',')
+conf.set10('ENABLE_GATEWAY_HOSTNAME', gateway_hostnames.contains('gateway'))
+conf.set10('ENABLE__GATEWAY_HOSTNAME', gateway_hostnames.contains('_gateway'))
+conf.set_quoted('GATEWAY_HOSTNAME',
+                conf.get('ENABLE__GATEWAY_HOSTNAME') == 1 ? '_gateway' : 'gateway',
+                description : 'synthetic gateway hostname as string')
```

Comment:
Hmm, i see now that this is basically what you implemented... but maybe still simplify this to become "--with-compatibility-gateway-hostname=BOOL" or so?